etc/scripts/freeze_and_update_reqs.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import fnmatch
+from commoncode.fileutils import resource_iter
+import os
+import subprocess
+import shutil
+import sys
+
+python_version = str(sys.version_info[0]) + str(sys.version_info[1])
+py_abi = "{0}cp{1}{0}".format("*", python_version)
+
+
+def generate_req_text(input_dir, output_file=False, package_name=False):
+    """
+    Generate a requirement.txt file of all dependencies present in thirdparty.
+    """
+    thirdparty = list(resource_iter(input_dir, with_dirs=False))
+    # FIXME this code is for py 3.6 and later we will update for all version
+    dependencies = [
+        files
+        for files in thirdparty
+        if fnmatch.fnmatchcase(files, "*py3*")
+        or fnmatch.fnmatchcase(files, py_abi)
+        or (
+            fnmatch.fnmatchcase(files, "*tar.gz*")
+            and not fnmatch.fnmatchcase(files, "*py2-ipaddress-3.4.1.tar.gz*")
+        )
+    ]
+    if not (os.path.isdir("required_deps")):
+        os.mkdir("required_deps")
+    for deps in dependencies:
+        shutil.copy(deps, "required_deps")
+    subprocess.run(
+        [
+            "pip-compile",
+            "--generate-hashes",
+            "--find-links",
+            "required_deps",
+            "--upgrade",
+            "--output-file",
+            output_file,
+            "--verbose",
+            "--upgrade-package",
```

Comment:
why do you do this? we never want to do some unwanted `upgrade-package` IMHO and what is `"package_name",` below about?

Reply:
this feature alow you to upgrade specfic package. See help by running this arg.

Reply:
I known that but what is `"package_name"` I know of no package whose name is `"package_name"` 
If this is some kind of magic, explain it in comments, do not make me chase down the help of that command

Reply:
In any case why would I want to upgrade a package there? Please elaborate and explain in comments

Reply:
Package name may be intbitset, banal, urlpy . Anything deps that exist in thirdparty

Reply:
You need to test this. The string `"package_name",` will NOT resolve to the variable `package_name`
`"package_name",` is of itself NOT the name a package...